src/DataStreams/PushingToViewsBlockOutputStream.cpp

```diff
@@ -386,14 +412,138 @@ void PushingToViewsBlockOutputStream::process(const Block & block, ViewInfo & vi
     catch (Exception & ex)
     {
         ex.addMessage("while pushing to view " + view.table_id.getNameForLogs());
-        view.exception = std::current_exception();
+        view.setException(std::current_exception());
     }
     catch (...)
     {
-        view.exception = std::current_exception();
+        view.setException(std::current_exception());
     }
 
-    view.elapsed_ms += watch.elapsedMilliseconds();
+    if (view.runtime_stats.thread_status)
+        view.runtime_stats.thread_status->updatePerformanceCounters();
+    view.runtime_stats.elapsed_ms += watch.elapsedMilliseconds();
 }
 
+void PushingToViewsBlockOutputStream::processPrefix(ViewInfo & view)
```

Comment:
Why do we need this method? Is it needed for setting a proper status for a view in a log? Because this code seems to be a copy paste with `processSuffix`.

Reply:
ProcessPrefix calls `writePrefix` to the view, while `processSuffix` calls `writeSuffix`. Aside from logs and error messages yes, they are 90% the same as most of the function is setup + teardown.

Reply:
Maybe better to simply add template function?

Reply:
It makes sense. Maybe even having a common runner for the 3 functions (prefix, blocks, suffix) that does the setup for the metrics, runs the function, and then the does the teardown.